Mathlib/AlgebraicGeometry/Spec.lean

```diff
@@ -112,100 +121,129 @@ theorem Spec.sheafedSpaceMap_id {R : CommRingCat} :
     NatTrans.ext _ _ <|
       funext fun U => by
         dsimp
-        erw [PresheafedSpace.id_c_app, comap_id]; swap
-        · rw [Spec.Top_map_id, TopologicalSpace.Opens.map_id_obj_unop]
-        simpa [eq_to_hom_map]
+        erw [NatTrans.comp_app, sheafedSpaceMap_c_app, PresheafedSpace.id_c_app, comap_id]; swap
+        · rw [Spec.topMap_id, TopologicalSpace.Opens.map_id_obj_unop]
+        simp [eqToHom_map]
+set_option linter.uppercaseLean3 false in
 #align algebraic_geometry.Spec.SheafedSpace_map_id AlgebraicGeometry.Spec.sheafedSpaceMap_id
 
 theorem Spec.sheafedSpaceMap_comp {R S T : CommRingCat} (f : R ⟶ S) (g : S ⟶ T) :
     Spec.sheafedSpaceMap (f ≫ g) = Spec.sheafedSpaceMap g ≫ Spec.sheafedSpaceMap f :=
   AlgebraicGeometry.PresheafedSpace.Hom.ext _ _ (Spec.topMap_comp f g) <|
-    NatTrans.ext _ _ <|
-      funext fun U => by dsimp; rw [CategoryTheory.Functor.map_id]; rw [category.comp_id];
-        erw [comap_comp f g]; rfl
+    NatTrans.ext _ _ <| funext fun U =>
```

Comment:
This deserves a porting note. It went from 3 lines to 13.

Reply:
I have made the proof shorter, this is an example where `dsimp` behaves very differently from its `Lean3` behaviour